### DISCUSSION TITLE
Turbo User getting Midroll ads

### DISCUSSION THREAD

**Discussion by FtrSpec** [streamlink/streamlink-twitch-gui#925] [Q&A] [answered]
With twitch changing things up (today), currently resulting in midroll streamlink ads.  Is there a way for me as a Turbo user (no ads all channels) to get no ads with streamlink?  The streamlink-twitch-gui has me enter and uses/verified my login but I'm still getting midroll ads (via streamlink).

**Comment by bastimeyer** [comment, marked as answer]
This doesn't belong here. See

1. https://streamlink.github.io/cli/plugins/twitch.html (docs are apparently slightly incorrect, see 4.)
2. https://github.com/streamlink/streamlink-twitch-gui/issues/923
3. https://github.com/streamlink/streamlink/issues/4934
4. https://github.com/streamlink/streamlink/issues/4939

**Reply by FtrSpec** [reply]
Thank you for that Authorization argument thread I'll have to try that out.
I would expect streamlink-twitch-gui to automatically pass the Authorization header.  I expect this to allow users to get timed twitch drops and remove ads for turbo users or channel subscribers.  If not automatically done (which I'd expect to be the default behavior) than perhaps consider a checkbox to pass credentials to the gui.

**Reply by bastimeyer** [reply]
> I would expect streamlink-twitch-gui to automatically pass the Authorization header.

As you can read in the docs from the first link, this is not possible. Twitch REST API != Twitch GQL API. Streaming access tokens can only be retrieved on the GQL API, and authorization/authentication happens there. The auth data is tied to the GQL API public key and logins do only work in your web browser. Streamlink Twitch GUI would need to access the regular Twitch.tv site and implement auth-flow spoofing by using one of the available Chromium extension APIs, which is 100% not going to happen.
